File: src/main/java/com/example/storage/controller/dto/EmployeeDto.java
package com.example.storage.controller.dto;

import com.example.storage.entity.Employee;
import com.example.storage.entity.Issue;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class EmployeeDto {
    private String name;
    private String department;
    private String photoUrl;
    private List<IssueDto> issues;

    public static EmployeeDto toDto(Employee employee) {
        List<IssueDto> issues = new ArrayList<>();
        for (Issue issue : employee.getIssueList()) {
            if (issue.getReturnedAt() == null) {
                issues.add(IssueDto.toDto(issue));
            }
        }

        return new EmployeeDto(employee.getName(), employee.getDepartment(), employee.getPhotoUrl(), issues);
    }
}

File: src/main/kotlin/ru/nto/storage/controller/dto/EmployeeDto.kt
package ru.nto.storage.controller.dto

import ru.nto.storage.entity.Employee

data class EmployeeDto(
    val name: String,
    val department: String?,
    val photoUrl: String?,
    val issues: List<IssueDto>
) {
    companion object {
        fun toDto(employee: Employee) = EmployeeDto(
            name = employee.name,
            department = employee.department,
            photoUrl = employee.photoUrl,
            issues = employee.issueList
                .filter { it.returnedAt == null }
                .map { IssueDto.toDto(it) }
        )
    }
}
